glossary/Higher-order-function.md

```diff
@@ -0,0 +1,3 @@
+### Higher-order function
+
+Higher-order functions or functors are functions that either take other functions as arguments, return a function as a result, or both.
```

Comment:
A Functor is a type. It specifically returns something that exposes a `map` method which when called will return another instance of the Functor on the supplied function. Functors are not always higher order functions, but they do return something that has a higher order function

```js
const Box = a => ({ map: fn => Box(fn(a)) })
```

Reply:
You can see more at Fantasy Land or in the Haskell type docs